subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/DefaultTaskCacheabilityResolver.java

```diff
@@ -64,13 +65,21 @@ public DefaultTaskCacheabilityResolver(RelativeFilePathResolver relativeFilePath
 
         for (OutputFilePropertySpec spec : outputFileProperties) {
             if (!(spec instanceof CacheableOutputFilePropertySpec)) {
-                return Optional.of(new CachingDisabledReason(
-                    CachingDisabledReasonCategory.NON_CACHEABLE_OUTPUT,
+                return Optional.of(new CachingDisabledReason(CachingDisabledReasonCategory.NON_CACHEABLE_OUTPUT,
                     "Output property '" + spec.getPropertyName() + "' contains a file tree"
                 ));
             }
         }
 
+        Optional<File> invalidRootFile = outputFileProperties.stream()
+            .flatMap(spec -> spec.getPropertyFiles().getFiles().stream())
+            .filter(file -> !file.exists()).findAny();
```

Comment:
should we use something more sophisticated than `.exists`, i.e. resolve it or check if it doesn't exist and is a symlink?

Reply:
I'm a bit confused about this, since if the task creates the symlink, how could it disable caching before running it, since the output is not yet a broken link.